src/components/GroupButtons.js

```diff
@@ -23,24 +23,25 @@ class CurrencyButton extends Component {
     return map[key];
   }
 
-  onCurrencyHandle(currency, time) {
-    if (currency === 'default') { currency = this.props.currency; } else { this.setState({ currencyActiveClass: currency }); }
-
-    if (time && time === 'END') {
-      this.setState({ exchangeRateValue: new Date(this.props.stats.time.endDate).yyyymmdd() });
-      this.setState({ exchangeRateActiveClass: time });
-      time = this.props.stats.time.endDate;
-    } else if (time && time === 'NOW') {
-      this.setState({ exchangeRateActiveClass: time });
-      this.setState({ exchangeRateValue: new Date().yyyymmdd() });
-    } else { time = this.state.exchangeRateValue; }
-
-    this.props.setCurrency(currency, time);
+  onCurrencyHandle(currency, dayClass) {
+    let rateDate = null;
+    switch (dayClass) {
+      case 'END':
+        rateDate = this.props.stats.time.endDate;
+        break;
+      default:
+        // includes NOW
+        rateDate = new Date();
+        break;
+    }
+    dayClass = dayClass || 'NOW';
+    currency = currency || this.props.currency;
```

Comment:
`dayClass = dayClass || 'NOW';`
This will take the default value as 'NOW' all the time, which not correct, we need to know what is the previous action for the user.
Solution:
move this in the top of the function, change the default value to be like this
`dayClass = dayClass || this.state.exchangeRateActiveClass || 'NOW';`